smart-hadoop-support/smart-hadoop-cdh-2.6/src/main/java/org/smartdata/hdfs/CompatibilityHelper26.java

```diff
@@ -122,4 +124,10 @@ public boolean truncate(DFSClient client, String src, long newLength) throws IOE
   public boolean truncate(DistributedFileSystem fs, String src, long newLength) throws IOException {
     throw new UnsupportedOperationException("Hadoop 2.6 does not support truncate.");
   }
+
+  @Override
+  public int getSidInDatanodeStorageReport(DatanodeStorage datanodeStorage) {
+    StorageType storageType = datanodeStorage.getStorageType();
```

Comment:
I remember that you said there is no 'StorageType' in Hadoop 2.6, so why do we have this?

Reply:
I have already found the 'StorageType' in hadoop 2.6. Hadoop 2.6 also has 'StorageType', but in different package.

Reply:
Good